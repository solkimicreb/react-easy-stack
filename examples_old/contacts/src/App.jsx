import React from "react";
import { view } from "react-easy-state";
import Contact from "./Contact";
import ContactCreator from "./ContactCreator";
import appStore from "./appStore";

// this rerenders whenever the appStore.contacts array changes (elements pushed or deleted)
function App() {
  return (
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Email</th>
          <th>Action</th>
        </tr>
      </thead>
      <tbody>
        {appStore.contacts.map(contact => (
          <Contact contact={contact} key={contact.email} />
        ))}
        <ContactCreator />
      </tbody>
    </table>
  );
}

// wrap the component with view() before exporting it
export default view(App);
