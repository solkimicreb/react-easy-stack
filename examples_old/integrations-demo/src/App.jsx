import React from "react";
import { view } from "react-easy-state";
import { params, path } from "react-easy-params";

const updateParam = ev => (params.name = ev.target.value);
const updatePath = ev => (path[0] = ev.target.value);

export default view(() => (
  <div>
    <div>
      Path: <input onChange={updatePath} value={path[0]} />
    </div>
    <div>
      Param: <input onChange={updateParam} value={params.name} />
    </div>
  </div>
));
