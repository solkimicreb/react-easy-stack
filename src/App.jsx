import React, { Component } from 'react'
import Router from './Router'
import Dummy from './Dummy'
import Link from './Link'

export default class App extends Component {
  async onRoute () {
    await new Promise(resolve => setTimeout(resolve, 1))
  }

  render () {
    return (
      <div>
        <Link to='/hello' element='li'>
          Link to Hello
        </Link>
        <Link to='/hi' element='li' params={{ name: 'Berci' }}>
          Link to Hi
        </Link>

        <Router>
          <Dummy page='hello' />
          <Router page='hi' default onRoute={this.onRoute}>
            <Dummy page='world' default />
            <Dummy page='you' />
          </Router>
        </Router>

        <Router>
          <Dummy page='hello' />
          <Dummy page='hi' default />
        </Router>
      </div>
    )
  }
}
