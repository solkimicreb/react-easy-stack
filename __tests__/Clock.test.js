import React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import sinon from 'sinon'
import App from '../examples/clock/App'

describe('Clock App', () => {
  const clock = sinon.useFakeTimers()
  const app = mount(<App />)
  const clearIntervalSpy = sinon.spy(global, 'clearInterval')

  afterAll(() => {
    clock.restore()
    clearIntervalSpy.restore()
  })

  test('should update to display the current time every second', async () => {
    expect(toJSON(app)).toMatchSnapshot('01. Initial state')

    clock.tick(2000)
    await Promise.resolve()
    expect(toJSON(app)).toMatchSnapshot('02. After 2 seconds')

    clock.tick(8500)
    await Promise.resolve()
    expect(toJSON(app)).toMatchSnapshot('03. After 10.5 seconds')
  })

  test('should clean up the interval timer when the component is unmounted', () => {
    app.unmount()
    expect(clearIntervalSpy.callCount).toBe(1)
  })
})
