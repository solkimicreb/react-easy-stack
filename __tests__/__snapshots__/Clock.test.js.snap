// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`01. Initial state 1`] = `
<App>
  <div>
    12:00:00 AM
  </div>
</App>
------------HTML PREVIEW---------------
<App>
  <div>
    12:00:00 AM
  </div>
</App>
`;

exports[`02. After 2 seconds 1`] = `
<App>
  <div>
    12:00:02 AM
  </div>
</App>
------------HTML PREVIEW---------------
<App>
  <div>
    12:00:02 AM
  </div>
</App>
`;

exports[`03. After 10.5 seconds 1`] = `
<App>
  <div>
    12:00:10 AM
  </div>
</App>
------------HTML PREVIEW---------------
<App>
  <div>
    12:00:10 AM
  </div>
</App>
`;
